File: 2006Scape Server/plugins/plugin/click/item/ItemOnNpc.java
package plugin.click.item;

import com.rs2.event.EventContext;
import com.rs2.event.EventSubscriber;
import com.rs2.event.SubscribesTo;
import com.rs2.event.impl.ItemOnNpcEvent;
import com.rs2.game.players.Player;

@SubscribesTo(ItemOnNpcEvent.class)
public final class ItemOnNpc implements EventSubscriber<ItemOnNpcEvent> {

    @Override
    public void subscribe(EventContext context, Player player, ItemOnNpcEvent event) {
        if (player.playerRights == 3) {
            player.getPacketSender().sendMessage("[ItemOnNpc] - itemId: " + event.getItem() + " npcId: " + event.getNpc());
        }
    }

}

File: 2006Scape Server/plugins/plugin/click/item/ItemOnNpc.kt
package plugin.click.item

import com.rs2.event.EventContext
import com.rs2.event.EventSubscriber
import com.rs2.event.SubscribesTo
import com.rs2.event.impl.ItemOnNpcEvent
import com.rs2.game.npcs.NpcHandler
import com.rs2.game.players.Player

@SubscribesTo(ItemOnNpcEvent::class)
class ItemOnNpc : EventSubscriber<ItemOnNpcEvent> {

	override fun subscribe(context: EventContext, player: Player, event: ItemOnNpcEvent) {

		if (player.playerRights >= 3) {
			player.packetSender.sendMessage("[ItemOnNpc] - itemId: ${event.item} npcId: ${event.npc}")
        }

	}
	
}